lib/utils/joinPath.ts

```diff
@@ -1,27 +1,37 @@
+import { NonEmptyArray } from './NonEmptyArray';
+
 /**
  * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#identifiers
  */
 const identifierRegex = /[$_\p{ID_Start}][$\u200c\u200d\p{ID_Continue}]*/u;
 
-export function joinPath(arr: Array<string | number>): string {
-  return arr.reduce<string>((acc, value) => {
+export function joinPath(path: NonEmptyArray<string | number>): string {
+  if (path.length === 1) {
+    return path[0].toString();
+  }
```

Comment:
note: the only real addition is this. Everything else is just a rename for readability.